flink-libraries/flink-state-processing-api/src/main/java/org/apache/flink/state/api/SavepointWriter.java

```diff
@@ -52,12 +54,42 @@
 @PublicEvolving
 public class SavepointWriter {
 
+    /**
+     * Loads an existing savepoint. Useful if you want to modify or extend the state of an existing
+     * application. The savepoint will be written using the state backend defined via the clusters
+     * configuration.
```

Comment:
What happens if the configuration doesn't (explicitly) specify any backend? I guess `HashMapStateBackend` will be used, right?

NIT: Should this be documented here and tested in `SavepointWriterTest`?  Or at least a reference in the javadoc?

ditto: `newSavepoint()`, `SavepointReader`.

Reply:
It will use whatever is configured in the flink-conf. So yes, HashMapStateBackend is the default.